proposals/0000-scan-takewhile-dropwhile.md

```diff
@@ -0,0 +1,138 @@
+# Feature name
+
+* Proposal: TBD
+* Author(s): [Kevin Ballard](https://github.com/kballard)
+* Status: **Awaiting review**
+* Review manager: TBD
+
+## Introduction
+
+Add 3 new `SequenceType` functions `scan(_:combine:)`, `takeWhile(_:)`, and
+`dropWhile(_:)`, with overrides as appropriate on `CollectionType`,
+`LazySequenceType`, and `LazyCollectionType`, as well as a global function
+`iterate(_:apply:)`.
+
```

Comment:
One of my concerns here is that while map, filter, and reduce are mostly
“lingua franca” among programmers everywhere, these others are much less
commonly-known and some of them aren't very descriptive of what they do.
There are lots more like these in the FP lexicon.  It was easy to draw
the line at map/filter/reduce (which also don't follow the usual grammar
for API guidelines), but if we take the ones in this proposal, where do
we stop?